pulser-core/pulser/register/_coordinates.py

```diff
@@ -24,41 +28,44 @@ class CoordsCollection:
         _coords: The coordinates.
     """
 
-    _coords: np.ndarray | list
+    _coords: pm.AbstractArray | list
```

Comment:
I would use just `AbstractArrayLike` instead

Reply:
If I make such change - a lot of `mypy` errors pop up saying that some types of `AbstractArrayLike` have no needed attributes, e.g., `Item "int" of "Union[Union[Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], AbstractArray], List[Any]]" has no attribute "__iter__" (not iterable)  [union-attr]`.

I think it's easier to leave `pm.AbstractArray` here.